src/test/java/bio/terra/service/dataset/DatasetServiceTest.java

```diff
@@ -93,16 +99,25 @@ public void setup() {
         // Setup mock sam service
         connectedOperations.stubOutSamCalls(samService);
         flightIdsList = new ArrayList<>();
+        datasetIdList = new ArrayList<>();
     }
 
     @After
     public void teardown() {
+        for (UUID datasetId : datasetIdList) {
+            datasetDao.delete(datasetId);
```

Comment:
What was happening without this before?

Reply:
There are deletes at the bottom of many of the tests, so it was working in the success case. My test didn't have the delete and had a failure first time through, so I added this.